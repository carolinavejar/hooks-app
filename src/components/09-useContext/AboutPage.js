import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const AboutPage = () => {
    const { user, setUser } = useContext(UserContext);

    const handleClick = ()=>{
        setUser({});
    }

    return (
        <div className="container">
            <h3>About Page</h3>
            <hr />

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>
            
            <button 
                className = "btn btn-primary"
                onClick = { handleClick }
            >
                Login
            </button>

        </div>
    )
}
