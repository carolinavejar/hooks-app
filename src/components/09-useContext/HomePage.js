import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomePage = () => {

    const userContext = useContext(UserContext);
    console.log(userContext);
    return (
        <div className="container">
            <h3>Home Pages</h3>
            <hr />
        </div>
    )
}
