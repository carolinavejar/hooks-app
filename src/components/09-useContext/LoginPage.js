import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginPage = () => {
    const { setUser } = useContext(UserContext);
    return (
        <div className="container">
            <h3>Login Screen</h3>
            <hr />
            <button 
                className = "btn btn-primary"
                onClick = { () => setUser({ name: 'Carito', id: 123 })}
            >
                Login
            </button>
        </div>
    )
}
