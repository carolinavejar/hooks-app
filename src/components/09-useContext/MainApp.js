import React from 'react'

export const MainApp = () => {
    return (
        <>
            <h3>Main App</h3>
            <hr />
        </>
    )
}
