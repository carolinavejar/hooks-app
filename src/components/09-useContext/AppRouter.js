import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { AboutPage } from './AboutPage';
import { LoginPage } from './LoginPage';
import { HomePage } from './HomePage';

export const AppRouter = () => {
    return (
        <Router>
            <>
                <Switch>
                    <Route exact path = "/" component = { HomePage } />
                    <Route exact path = "/about" component = { AboutPage } />
                    <Route exact path = "/login" component = { LoginPage } />
                </Switch>
            </>
        </Router>
    )
}
