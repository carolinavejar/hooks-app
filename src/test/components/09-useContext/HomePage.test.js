import React from 'react';
import { HomePage }  from '../../../components/09-useContext/HomePage';
import { UserContext } from '../../../components/09-useContext/UserContext';
import '@testing-library/jest-dom';

import { shallow, mount } from 'enzyme/build';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer( createSerializer ( { mode : 'deep' } ));

describe('Pruebas en el <HomePage /> ', () => {
    const user = {
        name: 'Carito',
        email: 'carito@gmail.com'
    };

    // shallow , renderiza solo primer componente
    // mount , renderiza todo en general 

    const wrapper = mount(
        <UserContext.Provider value = {{
            user
        }}>
              <HomePage/>
        </UserContext.Provider>
  
    );

    test('debe de mostrar <HomePage /> correctamente', () => {
        expect (wrapper).toMatchSnapshot();
    });

});
