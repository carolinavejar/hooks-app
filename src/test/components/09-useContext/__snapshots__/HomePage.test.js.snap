// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <HomePage />  debe de mostrar <HomePage /> correctamente 1`] = `
<div
  className="container"
>
  <h3>
    Home Pages
  </h3>
  <hr />
  <pre>
    {
   "name": "Carito",
   "email": "carito@gmail.com"
}
  </pre>
</div>
`;
